gui_scripts/gui_helpers/general_helpers.py

```diff
@@ -202,3 +204,210 @@ def stop(self):
             self.paused = False
             self.output_hints_signal.emit('Stopping simulation from thread')
         self.pause_condition.wakeOne()
+
+
+class DirectoryTreeView(QtWidgets.QTreeView):
+    """
+    Sets up a new directory tree view.
+    """
+    item_double_clicked_sig = QtCore.pyqtSignal(QtCore.QModelIndex)
+
+    def __init__(
+        self,
+        file_model: QtWidgets.QFileSystemModel,
+        parent: QtWidgets.QWidget = None
+    ):
+        super().__init__(parent)
+        self.setSelectionBehavior(QtWidgets.QTreeView.SelectRows)
+        self.setSelectionMode(QtWidgets.QTreeView.SingleSelection)
+
+        self.model = file_model
+        self.setModel(self.model)
+
+        self.copied_path = None
+        self.is_directory = False
+        self.is_cut_operation = False
+
+    def copy_item(
+        self,
+        source_index: QtCore.QModelIndex,
+        is_cut_operation: bool = False
+    ):
+        """
+        Implements the copy/cut operations in the directory tree view.
```

Comment:
I think sphinx needs a space between the description and the parameters but I’m not 100%, would you mind building and checking this too?

Reply:
I don't know if sphinx has anything on the gui. I ran
cd docs
make html
on both main and v1_1 and I didn't see anything in the docs for the gui